images/gateway/install.sh

```diff
@@ -0,0 +1,111 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+
+    printf "Usage: KCP_ORG=root:pipelines-service KCP_WORKSPACE=infra DATA_DIR=/workspace ./install.sh\n\n"
+
+    # Parameters
+    printf "The following parameters need to be passed to the script:\n"
+    printf "KCP_ORG: the organistation for which the workload clusters need to be registered, i.e.: root:pipelines-service\n"
+    printf "KCP_WORKSPACE: the name of the workspace where the gateway gets deployed (created if it does not exist), i.e: infra. If the workspace differs from the one where the WorkloadCluster has been created an APIBinding will need to be added\n"
+    printf "DATA_DIR: the location of the cluster files\n"
+}
+
+prechecks () {
+    KCP_ORG=${KCP_ORG:-}
+    if [[ -z "${KCP_ORG}" ]]; then
+        printf "KCP_ORG not set\n\n"
+        usage
+        exit 1
+    fi
+
+    KCP_WORKSPACE=${KCP_WORKSPACE:-}
+    if [[ -z "${KCP_WORKSPACE}" ]]; then
+        printf "KCP_WORKSPACE not set\n\n"
+        usage
+        exit 1
+    fi
+    
+    DATA_DIR=${DATA_DIR:-}
+    if [[ -z "${DATA_DIR}" ]]; then
+        printf "DATA_DIR not set\n\n"
+        usage
+	exit 1
+    fi
+}
+
+# populate kcp_kcfg with the location of the kubeconfig for connecting to kcp
+kcp_kubeconfig() {
+    if files=($(ls $DATA_DIR/credentials/kubeconfig/kcp/*.kubeconfig 2>/dev/null)); then
+        if [ ${#files[@]} -ne 1 ]; then
+            printf "A single kubeconfig file is expected at %s\n" "$DATA_DIR/credentials/kubeconfig/kcp"
+            usage
+            exit 1
+        fi
+        kcp_kcfg="${files[0]}"
+    else
+        printf "A single kubeconfig file is expected at %s\n" "$DATA_DIR/credentials/kubeconfig/kcp"
+        usage
+        exit 1
+    fi
+}
+
+switch_org() {
+    KUBECONFIG=${kcp_kcfg} kubectl kcp workspace use ${KCP_ORG}
+    if ! (KUBECONFIG=${kcp_kcfg} kubectl api-resources >> /dev/null 2>&1); then
+        printf "%s is not a valid organization, wrong kubectl context in use or connectivity issue\n" ${KCP_ORG}
+	usage
+	exit 1
+    fi
+}
+
+switch_ws() {
+    if (KUBECONFIG=${kcp_kcfg} kubectl get workspaces -o name | grep "${KCP_WORKSPACE}"); then
+        printf "use existing workspace\n"
+	KUBECONFIG=${kcp_kcfg} kubectl kcp workspace use ${KCP_WORKSPACE}
+
+    else
+       printf "creating workspace %s\n" "${KCP_WORKSPACE}"
+       KUBECONFIG=${kcp_kcfg} kubectl kcp workspace create "${KCP_WORKSPACE}" --enter
+    fi
+}
+
+install_gateway () {
+    KUBECONFIG=${kcp_kcfg}
+    CONFIG_DIR="${DATA_DIR}/environment/kcp/gateway"
+    KUBECONFIG=${kcp_kcfg} kubectl apply -k  "${CONFIG_DIR}"
+}
+
+main() {
+printf "Switching to organization %s\n" ${KCP_ORG}
+    prechecks
+    kcp_kubeconfig
+    printf "Switching to organization %s\n" ${KCP_ORG}
```

Comment:
I'd move the print statement in the function itself, but that might be a personal preference.